fvm/transactionFeeDeductor.go

```diff
@@ -37,14 +36,24 @@ func (d *TransactionFeeDeductor) Process(
 func (d *TransactionFeeDeductor) deductFees(
 	vm *VirtualMachine,
 	ctx *Context,
-	tx *flow.TransactionBody,
+	proc *TransactionProcedure,
 	sth *state.StateHolder,
 	programs *programs.Programs,
 ) (errors.Error, error) {
-	return vm.invokeMetaTransaction(
+
+	feeTx := deductTransactionFeeTransaction(proc.Transaction.Payer, ctx.Chain.ServiceAddress())
+	txErr, fatalErr := vm.invokeMetaTransaction(
 		*ctx,
-		deductTransactionFeeTransaction(tx.Payer, ctx.Chain.ServiceAddress()),
+		feeTx,
 		sth,
 		programs,
 	)
+	if txErr == nil {
+		proc.Events = append(proc.Events, feeTx.Events...)
+		proc.ServiceEvents = append(proc.ServiceEvents, feeTx.ServiceEvents...)
+		proc.Logs = append(proc.Logs, feeTx.Logs...)
```

Comment:
Shouldn't we collect this regardless?

Reply:
we can't right now, we don't have a way to discard some events, I'm working on an event collector handler with the functionality to purge some of the events similar to the state behavior. but right now this is consistent with how we capture events inside the transaction procedure.